### ISSUE TITLE
DB: Table Land Integration

### ISSUE THREAD

**Issue by Nasfame** [DeCenter-AI/decenter-ai.app#16]
- [ ]   Create users table with basic info

- name
- web3auth Id
- lastLoggedIn 
- createdAt
- updatedAt

- [ ]  Create table for model_train_requests

- id 
- userId
- modelName
- modelCID
- createdAt
- updatedAt

Caveat: work on william's repo

**Comment by techyNonso** [comment]
This is wrong,  when creating a user, you do not have access to the user's web3auth id. Else you want users to connect wallet when creating account

**Comment by Nasfame** [comment]
@techyNonso so web3auth id can be anything, it can be event user's public key: 

https://github.com/Web3Auth/web3auth-core-kit-examples/blob/main/node-sdk/node-backend-example/index.js

**Comment by Nasfame** [comment]
Here: https://web3auth.io/docs/integration-builder/builder/nextjs?lang=REACT&chain=ETH&evmFramework=WEB3&customAuth=NONE&mfa=DEFAULT&whitelabel=NO&useModal=YES&web3AuthNetwork=TESTNET&rnMode=EXPO&stepIndex=0


I think web3auth id would be in here:

```
  const getUserInfo = async () => {
    if (!web3auth) {
      uiConsole("web3auth not initialized yet");
      return;
    }
    const user = await web3auth.getUserInfo();
    uiConsole(user);
  };
```

**Comment by Nasfame** [comment]
Wow there is cooler nextjs snippet: https://web3auth.io/docs/integration-builder/builder/nextjs?lang=NEXT&chain=ETH&evmFramework=WEB3&customAuth=NONE&mfa=DEFAULT&whitelabel=NO&useModal=YES&web3AuthNetwork=TESTNET&rnMode=EXPO&stepIndex=0

**Comment by techyNonso** [comment]
You need to stop comparing your app to fully web3 app. On sign up form , there is no wallet connection so you cannot get any ID yet

**Comment by techyNonso** [comment]
> Here: https://web3auth.io/docs/integration-builder/builder/nextjs?lang=REACT&chain=ETH&evmFramework=WEB3&customAuth=NONE&mfa=DEFAULT&whitelabel=NO&useModal=YES&web3AuthNetwork=TESTNET&rnMode=EXPO&stepIndex=0
> 
> I think web3auth id would be in here:
> 
> ```
>   const getUserInfo = async () => {
>     if (!web3auth) {
>       uiConsole("web3auth not initialized yet");
>       return;
>     }
>     const user = await web3auth.getUserInfo();
>     uiConsole(user);
>   };
> ```

This function here will prompt for a wallet connection.

**Comment by techyNonso** [comment]
Mind you, a user can have more than one wallet

**Comment by Nasfame** [comment]
@Abhay-2811 its ready now

**Comment by Abhay-2811** [comment]
@Nasfame updating db on tableland costs gas are you sure you want to  update it every time user logs in ( talking about lastLoggedIn column )

**Comment by Nasfame** [comment]
Yes let's worth it.

**Comment by Abhay-2811** [comment]
making user_models's id auto-incrementing and integers ( 0,1,2.. such that )

**Comment by Abhay-2811** [comment]
Lets close this.. @Nasfame

**Comment by Nasfame** [comment]
Okay
